src/main/scala/edu/berkeley/ce/rockslicing/BlockVTK.scala

```diff
@@ -159,25 +158,40 @@ object BlockVTK {
     }
     offsetIterator(offsets, localOffsets.toSeq)
   }
+
+  /*
+   * FIXME: This is very inefficient and we probably want to use the right data
+   * structures to deal with duplicates for us.
+   */
+  private def distinctArrays(vertices: Seq[Array[Double]], previous: Seq[Array[Double]]=Nil): Seq[Array[Double]] = {
+    vertices match {
+      case Nil => previous
+      case vert+:verts =>
+        if (previous.exists(previousVertex => previousVertex sameElements vert)) {
+          distinctArrays(verts, previous)
+        } else {
+          distinctArrays(verts, vert+:previous)
+        }
+    }
+  }
 }
 
 /**
   * Simple data structure to contain data that represents a rock block that can be turned into vtk format
-  * @param Rock block 
   * @constructor Create a new rock block in format that can be turned into vtk by rockProcessor
   */
 case class BlockVTK(block: Block) {
   val orientedVertices = BlockVTK.orientVertices(block)
-  val tupleVertices = orientedVertices.values.flatten.toSeq.distinct
-  val connectivityMap = BlockVTK.connectivity(orientedVertices, tupleVertices)
-  val vertices = tupleVertices flatMap { t =>
-    List(t._1, t._2, t._3)
+  val arrayVertices = BlockVTK.distinctArrays(orientedVertices.values.flatten.toSeq)
```

Comment:
This operation will only be done once for each block if the `toVTK` flag is active. When all the calculations are done each `Block` will be used to create an equivalent `BlockVTK`. The smallest this array would be is 4 for a tetrahedron and somewhere in the 20's for a really complicated block. 12ish would be a good average where each entry is an `array` of length 3.

Based on this, I'd say using the faster approach with boxing might be better? At this point, we won't be generating any new elements through cutting and will only be calling this operation once per block. Does that make sense or would memory still be an issue when getting to really large numbers of blocks?